src/DataTable/DataTable.scss

```diff
@@ -299,12 +303,11 @@ $data-table-pagination-dropdown-min-width: 6rem !default;
 
   &.selection-right {
     > :first-child {
+      margin-right: map_get($spacers, 2);
+
       [dir="rtl"] & {
         margin-left: map_get($spacers, 2);
-      }
-
-      [dir="ltr"] & {
-        margin-right: map_get($spacers, 2);
```

Comment:
[clarification] This style wasn't applying in MFEs since the dev build doesn't compile SCSS/CSS with the same PostCSS RTL plugin as production builds.